### PR TITLE
docs(workflow): adds a description at the top of each file

### LINKED ISSUE #7607: Add short descriptions at the top of the workflow files
Zebra has quite a few workflow files: https://github.com/ZcashFoundation/zebra/tree/main/.github/workflows

As you browse that directory is pretty hard to know what each file do. Some of them have a short description at the top like https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/release-binaries.yml but the majority of them just don't:

- https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/continous-integration-docker.patch-always.yml
- https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/zcash-lightwalletd.yml
- https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/continous-delivery.yml
- ...

I think it will be nice to create a header doc that we can add to each of this files at the top with a short description on what can be expected in the code below.

I will be happy to do this myself but i unfortunately don't know what each file do.

### PR DESCRIPTION
**Pull request by gustavovalverde**
## Motivation

It's better if member of the team, existing or new contributors, can understand what a workflow file is meant to do (overall) before deep diving into it and trying to understand it piece by piece.

Closes #7607
Depends-On: #8009

#### Check before marking the PR as ready for review:
  - [X] Will the PR name make sense to users?
  - [X] Does the PR have a priority label?
  - [ ] Have you added or updated tests?
  - [X] Is the documentation up to date?

##### For significant changes:
  - [ ] Is there a summary in the CHANGELOG?
  - [ ] Can these changes be split into multiple PRs?

_If a checkbox isn't relevant to the PR, mark it as done._

## Solution

- Explain the workflow objective
- Explain some of the steps taken in the workflow, to accomplish the objetive

## Review

This PR depends on changes on the scrips, so I didn't had to make these changes twice.

### Reviewer Checklist

Check before approving the PR:
  - [x] Does the PR scope match the ticket?
  - [x] Are there enough tests to make sure it works? Do the tests cover the PR motivation?
  - [x] Are all the PR blockers dealt with?
        PR blockers can be dealt with in new tickets or PRs.

_And check the PR Author checklist is complete._

## Follow Up Work

None

### CODE REVIEW

**Comment by teor2345** [comment]
> Conflicting files
> .github/workflows/cd-deploy-nodes-gcp.yml
> .github/workflows/ci-integration-tests-gcp.yml
> .github/workflows/ci-unit-tests-docker.yml
> .github/workflows/docs-deploy-firebase.yml

Ah, sorry about that, I added descriptions as part of the external repository patch PRs, because it was starting to get really confusing.

File: .github/workflows/ci-build-crates.yml [hunk 1]
```diff
@@ -1,3 +1,13 @@
+# This workflow facilitates the individual building of Rust crates present in the repository.
+# 1. A matrix is generated dynamically to identify each crate in the repository.
+# 2. This matrix is checked for validity.
+# 3. Each identified crate undergoes three build processes:
+#    - With no default features.
```
Comment (upbqdn) [inline .github/workflows/ci-build-crates.yml]: ###### Optional Suggestion

```suggestion
#    - With no features.
```

**Review by upbqdn — APPROVED** [review]

**Review by upbqdn — APPROVED** [review]

**Comment by teor2345** [comment]
I think there are bugs in this branch that are fixed in the base PR, so I just updated it.

**Comment by teor2345** [comment]
Ah, that doesn't seem to work, maybe the bugs are still in the base PR.

**Pull request merged**